src/components/Searcher.js

```diff
@@ -982,8 +1020,7 @@ class Searcher extends React.Component<SearcherDefaultProps, SearcherProps, Sear
    * When a filter is removed, signal with weight 0 is created.
    */
   addFacetFilterSignal(facetFilter: FacetFilter, weight: number) {
```

Comment:
I would change the name of the parameter from `weight` to `added` or `removed` and make it be Boolean. Then you can set the weight on the signal object accordingly and it'll be semantically clearer.